src/SFA.DAS.AssessorService.Data/RegisterRepository.cs

```diff
@@ -77,29 +77,71 @@ public async Task<string>CreateEpaOrganisationStandard(EpaOrganisationStandard o
+                    "WHERE EndPointAssessorOrganisationId = @organisationId and standardCode = @standardCode";
+                var orgStandardId = await connection.ExecuteScalarAsync<int>(sqlToSelectExisting, new { organisationId = organisationStandard.OrganisationId, standardCode = organisationStandard.StandardCode });
 
-                var osdaId = connection.Query<string>(
-                    "INSERT INTO [dbo].[OrganisationStandard] ([EndPointAssessorOrganisationId],[StandardCode],[EffectiveFrom],[EffectiveTo],[DateStandardApprovedOnRegister] ,[Comments],[Status], [ContactId], [OrganisationStandardData]) VALUES (" +
-                    "@organisationId, @standardCode, @effectiveFrom, @effectiveTo, getutcdate(), @comments, 'Live', @ContactId,  @OrganisationStandardData); SELECT CAST(SCOPE_IDENTITY() as varchar); ",
-                    new
+                if(default(int) == orgStandardId)
+                {
+                    orgStandardId = connection.Query<int>(
+                        "INSERT INTO [dbo].[OrganisationStandard] ([EndPointAssessorOrganisationId],[StandardCode],[EffectiveFrom],[EffectiveTo],[DateStandardApprovedOnRegister] ,[Comments],[Status], [ContactId], [OrganisationStandardData], StandardReference) VALUES (" +
+                        "@organisationId, @standardCode, @effectiveFrom, @effectiveTo, getutcdate(), @comments, 'Live', @ContactId,  @OrganisationStandardData, @StandardReference); SELECT CAST(SCOPE_IDENTITY() as varchar); ",
+                        new
+                        {
+                            organisationStandard.OrganisationId,
+                            organisationStandard.StandardCode,
+                            organisationStandard.EffectiveFrom,
+                            organisationStandard.EffectiveTo,
+                            organisationStandard.DateStandardApprovedOnRegister,
+                            organisationStandard.Comments,
+                            organisationStandard.ContactId,
+                            organisationStandard.OrganisationStandardData,
+                            organisationStandard.StandardReference
+                        }).Single();
+
+                    foreach (var deliveryAreaId in deliveryAreas.Distinct())
                     {
-                        organisationStandard.OrganisationId, organisationStandard.StandardCode,
-                        organisationStandard.EffectiveFrom, organisationStandard.EffectiveTo,
-                        organisationStandard.DateStandardApprovedOnRegister, organisationStandard.Comments,
-                        organisationStandard.ContactId, organisationStandard.OrganisationStandardData
-                    }).Single();
+                        connection.Execute("INSERT INTO OrganisationStandardDeliveryArea ([OrganisationStandardId],DeliveryAreaId, Status) VALUES " +
+                                            "(@osdaId, @deliveryAreaId,'Live'); ",
```

Comment:
Bug here, `@osdaId` doesn't match the object named parameter being passed in, it throws an exception